src/features/preferences/renderer/preference-items/terminal/terminal-font-family/terminal-font-family.tsx

```diff
@@ -13,26 +13,29 @@ import { Select } from "../../../../../../renderer/components/select";
 import type { Logger } from "../../../../../../common/logger";
 import { action } from "mobx";
 import loggerInjectable from "../../../../../../common/logger.injectable";
+import {
+  terminalFontsInjectable,
+} from "../../../../../../renderer/components/dock/terminal/terminal-fonts.injectable";
 
 interface Dependencies {
   userStore: UserStore;
   logger: Logger;
+  terminalFonts: Map<string, string>;
 }
 
 const NonInjectedTerminalFontFamily = observer(
-  ({ userStore, logger }: Dependencies) => {
-
-    // fonts must be declared in `fonts.scss` and at `template.html` (if early-preloading required)
-    const supportedCustomFonts: SelectOption<string>[] = [
-      "RobotoMono", "Anonymous Pro", "IBM Plex Mono", "JetBrains Mono", "Red Hat Mono",
-      "Source Code Pro", "Space Mono", "Ubuntu Mono",
-    ].map(customFont => {
+  ({ userStore, logger, terminalFonts }: Dependencies) => {
+    const bundledFonts: SelectOption<string>[] = Array.from(terminalFonts.keys()).map(font => {
       const { fontFamily, fontSize } = userStore.terminalConfig;
```

Comment:
nit: this line could be moved to outside the `.map(() => {...})` which could simplify the code